uni/lib/model/providers/startup/profile_provider.dart

```diff
@@ -141,10 +150,10 @@ class ProfileProvider extends StateProviderNotifier<Profile> {
     final userPersistentInfo = PreferencesController.getPersistentUserInfo();
     if (userPersistentInfo != null) {
       final coursesDb = AppCoursesDatabase();
-      await coursesDb.saveNewCourses(profile.courses);
+      unawaited(coursesDb.saveNewCourses(profile.courses));
 
       final courseUnitsDatabase = AppCourseUnitsDatabase();
-      await courseUnitsDatabase.saveNewCourseUnits(allCourseUnits);
+      unawaited(courseUnitsDatabase.saveNewCourseUnits(allCourseUnits));
```

Comment:
I believe this can lead to invalid caches due to saveNewCourses and saveNewCourseUnits not being atomic operations, but it is better to address this at the database level.

Reply:
Yeah, should be done at DB level. Insert as far as I remember are already atomic, not sure about entire transactions though. Feel free to open an issue